providers/statsig/pkg/provider.go

```diff
@@ -0,0 +1,320 @@
+		ProviderResolutionDetail: res.ProviderResolutionDetail,
+	}
+}
+
+func (p *Provider) IntEvaluation(ctx context.Context, flag string, defaultValue int64, evalCtx of.FlattenedContext) of.IntResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.IntResolutionDetail{
+		Value:                    res.Value.(int64),
+		ProviderResolutionDetail: res.ProviderResolutionDetail,
+	}
+}
+
+func (p *Provider) StringEvaluation(ctx context.Context, flag string, defaultValue string, evalCtx of.FlattenedContext) of.StringResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.StringResolutionDetail{
+		Value: fmt.Sprint(res.Value),
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			ResolutionError: res.ProviderResolutionDetail.ResolutionError,
+			Reason:          res.ProviderResolutionDetail.Reason,
+			Variant:         res.Variant,
+			FlagMetadata:    res.FlagMetadata,
+		},
+	}
+}
+
+func (p *Provider) ObjectEvaluation(ctx context.Context, flag string, defaultValue interface{}, evalCtx of.FlattenedContext) of.InterfaceResolutionDetail {
+
+	// TODO to be removed on new SDK version adoption which includes https://github.com/open-feature/spec/issues/238
+	if p.status != of.ReadyState {
+		if p.status == of.NotReadyState {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewProviderNotReadyResolutionError(providerNotReady),
+					Reason:          of.ErrorReason,
+				},
+			}
+		} else {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	statsigUser, err := toStatsigUser(evalCtx)
+	if err != nil {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError(err.Error()),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	featureConfig, err := toFeatureConfig(evalCtx)
+	if err != nil {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError(err.Error()),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+	if featureConfig.FeatureConfigType == CONFIG {
+		config := statsig.GetConfig(*statsigUser, featureConfig.Name)
+		flagMetadata := make(map[string]interface{})
+		flagMetadata["GroupName"] = config.GroupName
+		flagMetadata["LogExposure"] = config.LogExposure
+		flagMetadata["Name"] = config.Name
+		flagMetadata["RuleID"] = config.RuleID
+		return of.InterfaceResolutionDetail{
+			Value: config.Value[flag],
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				FlagMetadata: flagMetadata,
+			},
+		}
+	} else {
```

Comment:
I think an explicit check for `LAYER` is better here

Reply:
done